plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -2375,7 +2389,16 @@ def rsync_destination_container(gear, destination_container, destination_distric
       reply = ResultIO.new
       source_container = gear.get_proxy
       platform = gear.group_instance.platform
+      gear_quota = get_quota(gear)
       log_debug "DEBUG: Gear platform is '#{platform}'"
+      destination_available_space = destination_container.get_node_disk_free
+
+      # check here to make sure addition of gear to destination_container
+      # will not result in > 98% of available free space on destination_container
+      if gear_quota[1].to_i > 0.98 * destination_available_space
```

Comment:
It's not obvious from the name what `gear_quota[1]` is, and you use it twice, so a descriptively named local variable (e.g., `gear_blocks_used`) might be in order.